### PR TITLE
PIL-1070: Local fiat currency changing doesn't work

### PR DESCRIPTION
**Pull request by mdjastrzebski**
Scope:
- [x] Support fiat currencies for total balances (by memoized selectors)
- [x] Support fiat currencies for asset balances (by in component handling)
- [x] Some minor refactor regarding to totalBalances type naming
- [x] Strongly typed Currency usage

### CODE REVIEW

File: src/models/TotalBalances.js [hunk 1]
```diff
@@ -20,41 +20,46 @@
...
-  rewards?: T,
-};
-
-export type StoreTotalBalances = AccountRecord<?StoreAccountTotalBalances>;
-export type StoreAccountTotalBalances = StoreCategoryRecord<ChainRecord<BigNumber>>;
-
-// Wallet balances are calculated using selectors in redux
-export type WalletTotalBalances = AccountRecord<ChainRecord<BigNumber>>;
+/** Generic helper type for Account -> Category -> Chain nesting. */
+export type AccountCategoryChainRecord<T> = AccountRecord<AssetCategoryRecord<ChainRecord<T>>>;
```
Comment (poocart) [inline src/models/TotalBalances.js:38]: 😱

Comment (mdjastrzebski) [reply src/models/TotalBalances.js:38]: Yup, this is our actual data structure, only renamed it for clarity

File: src/utils/rates.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  const ethRates = rates[ETH];
+  if (!ethRates || !ethRates[currency] || !ethRates[USD]) {
+    return null;
+  }
+
+  return ethRates[currency] / ethRates[USD];
+}
+
+export function getFiatValueFromUsd(valueInUsd: ?BigNumber | string, usdToFiatRate: ?number): ?BigNumber {
+  return valueInUsd && usdToFiatRate != null ? BigNumber(valueInUsd).times(usdToFiatRate) : null;
```
Comment (poocart) [inline src/utils/rates.js:41]: Isn't it BigNumber already?

```suggestion
  return valueInUsd && usdToFiatRate != null ? valueInUsd.times(usdToFiatRate) : null;
```

**Review by poocart — APPROVED** [review]
LGTM 🚀

File: src/utils/rates.js (hunk 2 above)
Comment (mdjastrzebski) [reply src/utils/rates.js:41]: When coming back from redux restore it can be string

Comment (poocart) [reply src/utils/rates.js:41]: Nevermind, maybe having it wrapped here makes sense, but maybe we can use `wrapBigNumber` instead, wdyt?

Comment (mdjastrzebski) [reply src/utils/rates.js]: Applied

**Pull request merged**